components/otter_waiver/sources/common/base.mjs

```diff
@@ -0,0 +1,66 @@
+// import crypto from "crypto";
+import otterWaiver from "../../otter_waiver.app.mjs";
+
+export default {
+  type: "source",
+  dedupe: "unique",
+  props: {
+    otterWaiver,
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+    db: "$.service.db",
+  },
+  methods: {
+    _getWebhookId() {
+      return this.db.get("webhookId");
+    },
+    _setWebhookId(id) {
+      this.db.set("webhookId", id);
+    },
+    emitEvent(event) {
+      this.$emit(event, {
+        id: event.id,
+        summary: this.getSummary(event),
+        ts: Date.parse(event.timeStamps.checkedIn),
```

Comment:
Based on the test-events, it looks like `checkedIn` can be null. Probably safer to use `Date.now()`.
```suggestion
        ts: Date.now(),
```